ddtrace/profiling/collector/_lock.py

```diff
@@ -263,42 +263,36 @@ def _find_self_name(self, var_dict: typing.Dict):
         return None
 
     # Get lock acquire/release call location and variable name the lock is assigned to
+    # This function propagates ValueError if the frame depth is <= 3.
     def _maybe_update_self_name(self):
         if self._self_name is not None:
             return
-        try:
-            # We expect the call stack to be like this:
-            # 0: this
-            # 1: _acquire/_release
-            # 2: acquire/release (or __enter__/__exit__)
-            # 3: caller frame
-            if config.enable_asserts:
-                frame = sys._getframe(1)
-                if frame.f_code.co_name not in {"_acquire", "_release"}:
-                    raise AssertionError("Unexpected frame %s" % frame.f_code.co_name)
-                frame = sys._getframe(2)
-                if frame.f_code.co_name not in {
-                    "acquire",
-                    "release",
-                    "__enter__",
-                    "__exit__",
-                    "__aenter__",
-                    "__aexit__",
-                }:
-                    raise AssertionError("Unexpected frame %s" % frame.f_code.co_name)
-            frame = sys._getframe(3)
-
-            # First, look at the local variables of the caller frame, and then the global variables
-            self._self_name = self._find_self_name(frame.f_locals) or self._find_self_name(frame.f_globals)
-
-            if not self._self_name:
-                self._self_name = ""
-                LOG.debug(
-                    "Failed to get lock variable name, we only support local/global variables and their attributes."
-                )
-
-        except Exception as e:
-            LOG.warning("Error getting lock acquire/release call location and variable name: %s", e)
+        # We expect the call stack to be like this:
+        # 0: this
+        # 1: _acquire/_release
+        # 2: acquire/release (or __enter__/__exit__)
+        # 3: caller frame
+        if config.enable_asserts:
+            frame = sys._getframe(1)
+            if frame.f_code.co_name not in {"_acquire", "_release"}:
+                raise AssertionError("Unexpected frame %s" % frame.f_code.co_name)
+            frame = sys._getframe(2)
+            if frame.f_code.co_name not in {
+                "acquire",
+                "release",
+                "__enter__",
+                "__exit__",
+                "__aenter__",
+                "__aexit__",
+            }:
+                raise AssertionError("Unexpected frame %s" % frame.f_code.co_name)
```

Comment:
Is raising an error the standard pattern here rather than just logging the error and continuing